app/src/main/java/ca/unb/mobiledev/reflexrevolution/SwipeActivity.java

```diff
@@ -0,0 +1,141 @@
+package ca.unb.mobiledev.reflexrevolution;
+
+import android.os.Bundle;
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+import androidx.appcompat.app.AppCompatActivity;
+import java.util.Random;
+
+//Author: Hailey Savoie
+//Ideas taken from: https://riptutorial.com/android/example/16543/swipe-detection
+
+public class SwipeActivity extends AppCompatActivity {
+
+    private Random r;
+    private int index;
+    private int score;
+    private TextView swipeText;
+    private TextView scoreText;
+    private String[] directions;
+    private SwipeListener swipeListener;
+    private RelativeLayout relativeLayout;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState){
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.swipe_activity);
+
+        relativeLayout = findViewById(R.id.relative_layout);
+        swipeText = findViewById(R.id.text_view);
+        scoreText = findViewById(R.id.text_view2);
+        swipeListener = new SwipeListener(relativeLayout);
+
+        score = 0;
+        r = new Random();
+        directions = new String[]{"right", "left", "up", "down"};
+
+        newDirection();
+    }
+
+    private void newDirection(){
+        index = r.nextInt(4 - 0) + 0;
+        swipeText.setText("Swipe " + directions[index]);
+    }
+
+    private void updateScore(){
+        scoreText.setText("Score: " + score);
+    }
+
+    private class SwipeListener implements View.OnTouchListener{
```

Comment:
Should I make this a separate class?

Reply:
Yes, it will need to be a separate class since that is what we will use to integrate it with the main game loop.